irange-ify

r-barnes opened the pull request:
Summary:
Auto-generated with D28874212

Per D25969148, the number of modernizable loops is fairly small

Differential Revision: D29834756

malfet commented on torch/csrc/jit/tensorexpr/operators/reduction.cpp:
```diff
@@ -116,7 +117,7 @@ Tensor* computeMean(
     // NOLINTNEXTLINE(clang-diagnostic-sign-compare)
-    for (int64_t idx = 0; idx < InputBuf.ndim(); idx++) {
+    for (const auto idx : c10::irange(InputBuf.ndim())) {
       mean_dims_expr.emplace_back(idx);
```
Hmm, you'll need to explicitly typecast `idx` to one of the types defined in 
https://github.com/pytorch/pytorch/blob/d6e776d96110a3b0c7d4777b03e4c400ae9a46eb/c10/core/ScalarType.h#L125-L131

r-barnes committed:
irange-ify (#62045)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/62045

Auto-generated with D28874212

Per D25969148, the number of modernizable loops is fairly small

Differential Revision: D29834756

fbshipit-source-id: 4aaa5fbe39c37e7f4f96e741265122867c1dc6a5
